.github/workflows/publish.yaml

```diff
@@ -2,8 +2,8 @@ name: Build and Publish Package to Pypi
 
 on:
   push:
-    branches:
-        - "main"
+    tags:
+      - "*"
```

Comment:
I do not think this is what we want. We need to enforce that only for those tags created in the `main` branch do we publish a new version to `pypi`. Tags on other branches should to be ignored, until they are merged to main.

Reply:
> I do not think this is what we want. We need to enforce that only for those tags created in the `main` branch do we publish a new version to `pypi`. Tags on other branches should to be ignored, until they are merged to main.

Not sure if I get it. What are "Tags on other branches"?

Reply:
I am not too sure about this, but as far as I understand tags are on commits. What I meant was that if the corresponding commit is not in the `main` branch, for example someone created a `just_for_fun` tag on his/her own dev branch, we should not publish a new version to `pypi` for this.

Reply:
> I am not too sure about this, but as far as I understand tags are on commits. What I meant was that if the corresponding commit is not in the `main` branch, for example someone created a `just_for_fun` tag on his/her own dev branch, we should not publish a new version to `pypi` for this.

To leave out this kind of `just_for_fun` tags, we might just create a regex match condition for the tag versions that trigger the workflow.

I created a sample workflow.
https://github.com/kausalflow/hamilflow/issues/41

Reply:
This does not help essentially again my argument. one can also create a tag `10.100.1000` on his / her dev branch, for any reason (e.g. for experiment).

Reply:
> This does not help essentially against my argument. one can also create a tag `10.100.1000` on his / her dev branch, for any reason (e.g. for experiment).

Ah, I got your point. However, I am not sure how to do it in Actions. Any suggestions?

Reply:
Let us leave an issue for this and move forward for now.